### PR TITLE
feat: add metrics for provider

### LINKED ISSUE #400: Add metrics for provider requests
**Describe the solution you'd like**
[A clear and concise description of what you want to happen.]
We have metrics today for the csi driver. Now that the provider runs as a standalone pod, we can enable metrics for the provider.

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Secrets Store CSI Driver version: (use the image tag):
- Azure Key Vault provider version: (use the image tag):
- Kubernetes version: (use `kubectl version`):
- Cluster type: (e.g. AKS, aks-engine, etc):

### PR DESCRIPTION
**Pull request by aramase**
Signed-off-by: Anish Ramasekar <anish.ramasekar@gmail.com>

**Reason for Change**:

- Adds metrics framework with otel

**Requirements**

- [x] squashed commits
- [x] included documentation
- [x] added unit tests and e2e tests (if applicable).

**Issue Fixed**:

fixes #400 

**Does this change contain code from or inspired by another project?**

- [ ] Yes
- [ ] No

**If "Yes," did you notify that project's maintainers and provide attribution?**

**Special Notes for Reviewers**:

### CODE REVIEW

File: cmd/main.go [hunk 1]
```diff
@@ -106,9 +112,10 @@ func main() {
 		grpc.UnaryInterceptor(utils.LogGRPC),
```
Comment (nilekhc) [inline cmd/main.go:112]: Can we add [metrics for grpc](https://github.com/Azure/kubernetes-kms/blob/c3180993ec313a0fe18228af10f384c42f35dcb0/cmd/server/main.go#L90) as well?

Comment (aramase) [reply cmd/main.go]: Added metrics for gRPC too. The gRPC request has a different set of attributes compared to the keyvault metric, so I've kept that as a separate metric. The keyvault metric is what will be useful for users to monitor the plugin is functioning correctly.

File: pkg/utils/grpc.go [hunk 2]
```diff
@@ -21,14 +23,21 @@ func ParseEndpoint(ep string) (string, string, error) {
...
+		start := time.Now()
+		reporter := metrics.NewStatsReporter()
+
+		ctxDeadline, _ := ctx.Deadline()
+		klog.V(5).InfoS("request", "method", info.FullMethod, "deadline", time.Until(ctxDeadline).String())
+
+		resp, err := handler(ctx, req)
+		s, _ := status.FromError(err)
+		klog.V(5).InfoS("response", "method", info.FullMethod, "duration", time.Since(start).String(), "code", s.Code().String(), "message", s.Message())
+		reporter.ReportGRPCRequest(ctx, time.Since(start).Seconds(), info.FullMethod, s.Code().String(), s.Message())
```
Comment (aramase) [inline pkg/utils/grpc.go:39]: Using the full method for metric to track the proto version for the RPC call

**Review by nilekhc — APPROVED** [review]

**Pull request merged**